backend/code/archipelag/market/views.py

```diff
@@ -23,22 +24,41 @@ def get(self, request):
 @login_required
 def market_create(request):
     if request.method == 'POST':
-        body_unicode = request.body.decode('utf-8')
-        body_data = loads(body_unicode)
-
-        new_market = Market.objects.create(
-            owner=request.user.ngouser,
-            title=body_data["title"],
-            url=body_data["url"],
-            date_starting=body_data["date_starting"],
-            date_ending=body_data["date_ending"],
-            hashtag = body_data["hashtag"]
-        )
-        data = dict(url='/message/create/%s'%new_market.id)
-        return JsonResponse(data)
+        body_data = loads(request.body.decode('utf-8'))
+        current_ngo = request.user.ngouser
+        if can_create_market(current_ngo):
+            delete_points(current_ngo)
+            new_market = get_new_market(current_ngo, body_data)
+            data = dict(url='/message/create/%s'%new_market.id)
+            return JsonResponse(data)
+        else:
+            error = dict(error="Za mało punktów.")
+            return JsonResponse(error)
     return render(request, 'registration/event.html', )
 
 
+def get_new_market(current_ngo, body_data):
+    new_market = Market.objects.create(
+        owner=current_ngo,
+        title=body_data["title"],
+        url=body_data["url"],
+        date_starting=body_data["date_starting"],
+        date_ending=body_data["date_ending"],
+        hashtag=body_data["hashtag"]
+    )
+    return new_market
+
+
+def can_create_market(current_ngo):
```

Comment:
I do not understand. You need to create new NgoUser to check something relative to the current_ngo? Why you can not check it from current_ngo?

Reply:
You are right I should get this method from current_ngo, thanks